src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -204,12 +205,22 @@ private[snowflake] class JDBCWrapper {
             throw new IllegalArgumentException(s"Don't know how to save $field of type ${field.name} to Snowflake")
         }
         val nullable = if (field.nullable) "" else "NOT NULL"
-        sb.append(s""", ${name.replace("\"", "\\\"")} $typ $nullable""".trim)
+        sb.append(s""", ${if (isQuoted(name)) name else quotedName(name)} $typ $nullable""".trim)
```

Comment:
gotcha! Also modified the test case in one of the unit tests, `SnowflakeSourceSuite.scala:324-325` and added in another case that ensures explicitly quoted inputs are not uppercased in `SnowflakeSourceSuit.scala:329`